src/wattsi.pas

```diff
@@ -904,6 +904,59 @@    TCrossReferences = record
       end;
    end;
 
+   procedure InsertMDNAnnotationForElement(const Element: TElement);
+   var
+      Candidate: TNode;
+      ID: UTF8String;
+      TargetAncestor, MDNBox: TElement;
+   begin
+      if (not(Element.HasAttribute('id'))) then
+         exit;
+      ID := Element.GetAttribute('id').AsString;
+      if (not(MDNJSONData[ID] is TJSONArray)) then
+         // No MDN article has a link to this ID.
+         exit;
+
+      MDNBox := E(eAside, ['class', 'mdn before wrapped']);
+
+      if (Element.HasProperties(propHeading)) then
+      begin
+         // If this is a heading, we insert the annotation after the element
+         // being annotated. When generating multipage output, inserting the
+         // annotation after the heading ensures that it ends up in the right
+         // file. It can otherwise incorrectly end up in the previous split.
+         MDNBox := E(eAside, ['class', 'mdn wrapped']);
+         if ((Element.NextSibling is TElement)
+               and (TElement(Element.NextSibling)
+                  .GetAttribute('class').AsString = 'mdn wrapped')) then
+            // If there's already an MDN box at the point where we want this,
+            // then just re-use it (instead of creating another one).
+            MDNBox := TElement(Element.NextSibling)
```

Comment:
Use NextElementSibling? Otherwise intervening whitespace will block the consolidation, right?

Reply:
I had reverted the entire commit wholesale that had included the addition of NextElementSibling. But I now un-reverted the part of the patch that changed dom.pas, so we now have NextElementSibling, and I changed the code here to use it